### PR TITLE
Fixes for PatchTST Config

### PR DESCRIPTION
**Pull request by wgifford**
# What does this PR do?

Fixes a bug in the handling of the `num_patches` parameter which was removed from the config.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
fee free to ping me or @kashif when this is ready for review 😉

**Review by kashif — APPROVED** [review]

File: src/transformers/models/patchtst/modeling_patchtst.py [hunk 1]
```diff
@@ -1736,6 +1738,7 @@ def forward(
             past_observed_mask=past_observed_mask,
             output_hidden_states=output_hidden_states,
             output_attentions=output_attentions,
+            return_dict=True,
```
Comment (ArthurZucker) [inline src/transformers/models/patchtst/modeling_patchtst.py:1741]: why do we enforce this?

File: src/transformers/models/patchtst/modeling_patchtst.py [hunk 2]
```diff
@@ -1962,10 +1965,10 @@ def forward(
             past_observed_mask=past_observed_mask,
             output_hidden_states=output_hidden_states,
             output_attentions=output_attentions,
-            return_dict=return_dict,
+            return_dict=True,
         )
         # get output head. y_hat is of shape [bs x num_targets] or tuple of this shape
-        y_hat = self.head(model_output[0])
+        y_hat = self.head(model_output.last_hidden_state)
```
Comment (ArthurZucker) [inline src/transformers/models/patchtst/modeling_patchtst.py:1971]: mmm this seems like a regression to me ?

**Review by ArthurZucker — COMMENTED** [review]
patch changes look good but we should keep the return dict logic IMO

File: src/transformers/models/patchtst/modeling_patchtst.py (hunk 2 above)
Comment (kashif) [reply src/transformers/models/patchtst/modeling_patchtst.py:1971]: so my question was if the call to the model is internal, can we set the dict to something convenient and then output the dict as per the forward or shall we pass the return dict to the internal model?

Comment (wgifford) [reply src/transformers/models/patchtst/modeling_patchtst.py:1971]: @kashif I think it is much clearer to use `last_hidden_state` while enforcing `return_dict` in the internal call. Externally, the `forward` method still honors the `return_dict` passed in by the caller.

Comment (ArthurZucker) [reply src/transformers/models/patchtst/modeling_patchtst.py:1971]: Fine with me, but then the class should not even support return dict no?

Comment (ArthurZucker) [reply src/transformers/models/patchtst/modeling_patchtst.py:1971]: Ok

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**